lib/index.js

```diff
@@ -48,14 +48,27 @@ function getRepo(options) {
  * Push a git branch to a remote (pushes gh-pages by default).
  * @param {string} basePath The base path.
  * @param {Object} config Publish options.
- * @param {Function} done Callback.
+ * @param {Function} callback Callback.
  */
-exports.publish = function publish(basePath, config, done) {
+ /*eslint-disable no-console */
+exports.publish = function publish(basePath, config, callback) {
   if (typeof config === 'function') {
-    done = config;
+    callback = config;
     config = {};
   }
 
+  var done = function(arguments) {
+    try {
+      (callback || function(err) {
+        if (!err) {
+          console.log(err)
```

Comment:
Yes.
I think if the caller is not interested in handling any successful callback, because it's the last action anyway or so, then he should be notified in the error case.

But you can also omit the logging in the binary here https://github.com/tschaub/gh-pages/blob/master/bin/gh-pages#L35
and do a successful log in this file.